include/alpaka/core/ConcurrentExecPool.hpp

```diff
@@ -159,7 +159,7 @@ namespace alpaka
                 TaskPkg(
                     TFnObj && func) :
                         m_Promise(),
-                        m_FnObj(std::forward<TFnObj>(func))
+                        m_FnObj(func)
```

Comment:
The same as above should be true here. Therefore we should use `typename std::remove_reference<TFnObj>::type m_FnObj;` as the member to allow only copies and moves as the current version (even without forward) allows to hold a reference.